results/cmd/api/main_test.go

```diff
@@ -118,37 +119,286 @@ func TestUpdateTaskRun(t *testing.T) {
+				},
+			},
+			expect: &pb.TaskRun{
+				ApiVersion: "v1alpha1",
+				Metadata: &pb.ObjectMeta{
+					Uid:       "31415926",
+					Name:      "mytaskrun",
+					Namespace: "default",
+				},
+			},
+		},
+		{
+			in: &pb.TaskRun{
+				ApiVersion: "v1beta1",
+				Metadata: &pb.ObjectMeta{
+					Uid:       "31415926",
+					Name:      "mytaskrun",
+					Namespace: "default",
+				},
+				Spec: &pb.TaskRunSpec{
+					TaskSpec: &pb.TaskSpec{
+						Steps: []*pb.Step{
+							{
+								Name: "first_step",
+							},
+							{
+								Name: "next_step",
+							},
+						},
+					},
+				},
+			},
+			name:      "Test Mask updating repeated fields",
+			fieldmask: &field_mask.FieldMask{Paths: []string{"spec.task_spec.steps"}},
+			// update entire repeated field(all elements in array) - standard update
+			update: &pb.TaskRun{
+				ApiVersion: "v1beta1",
+				Metadata: &pb.ObjectMeta{
+					Uid:       "31415926",
+					Name:      "mytaskrun",
+					Namespace: "default",
+				},
+				Spec: &pb.TaskRunSpec{
+					TaskSpec: &pb.TaskSpec{
+						Steps: []*pb.Step{
+							{
+								Name: "new_first_step",
+							},
+							{
+								Name: "new_next_step",
+							},
+						},
+					},
+				},
+			},
+			expect: &pb.TaskRun{
+				ApiVersion: "v1beta1",
+				Metadata: &pb.ObjectMeta{
+					Uid:       "31415926",
+					Name:      "mytaskrun",
+					Namespace: "default",
+				},
+				Spec: &pb.TaskRunSpec{
+					TaskSpec: &pb.TaskSpec{
+						Steps: []*pb.Step{
+							{
+								Name: "new_first_step",
+							},
+							{
+								Name: "new_next_step",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			in: &pb.TaskRun{
+				ApiVersion: "v1alpha1",
+				Metadata: &pb.ObjectMeta{
+					Uid:       "31415926",
+					Name:      "mytaskrun",
+					Namespace: "default",
+				},
+			},
+			name:      "Test Mask with nil Paths field",
+			fieldmask: &field_mask.FieldMask{},
+			// do not update
+			update: &pb.TaskRun{
+				ApiVersion: "v1beta1",
+				Metadata: &pb.ObjectMeta{
+					Uid:       "123456",
+					Name:      "newtaskrun",
+					Namespace: "tekton",
+				},
+			},
+			expect: &pb.TaskRun{
+				ApiVersion: "v1alpha1",
+				Metadata: &pb.ObjectMeta{
+					Uid:       "31415926",
+					Name:      "mytaskrun",
+					Namespace: "default",
+				},
+			},
+		},
 	}
-	for rows.Next() {
-		var taskrunblob []byte
-		taskrun := &pb.TaskRun{}
-		rows.Scan(&taskrunblob)
-		if err := proto.Unmarshal(taskrunblob, taskrun); err != nil {
-			t.Fatal("unmarshaling error: ", err)
-		}
-		if diff := cmp.Diff(taskrun.String(), r.GetTaskRun().String()); diff != "" {
-			t.Fatalf("Update Function not properly implemented: %v", diff)
-		}
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			r, err := srv.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{TaskRun: tc.in})
+			if err != nil {
+				t.Fatalf("could not create taskrun: %v", err)
+			}
+			// Update the created taskrun
+			fieldMask := tc.fieldmask
+			if tc.name == "Test no Mask" {
```

Comment:
Now that you're passing in a pointer, you should be able to omit this.

Reply:
done